### ISSUE TITLE
An inference error occurred after converting ONNX to tensorrt

### ISSUE THREAD

**Issue by wangbiao0** [NVIDIA/TensorRT#4460]
When I was using UNet for image segmentation, after converting the pytorch format model to the tensorrt format, there would be a large error in the reasoning, as follows:  

![Image](https://github.com/user-attachments/assets/f9c63f4b-112c-4d62-a0f1-c9ccb945df2b)  

But the label image is like the following:  

![Image](https://github.com/user-attachments/assets/f3512284-d563-4312-ac80-1b73195d48fc)

However, when I perform reasoning using the.pth format or the.onnx format, I can obtain a normal result (the prediction result is similar to a label).   

When converting the onnx model to tensorrt format, I tried various methods, using code and tools (trtexec and Polygraphy), and employed a wide range of precisions (FP32/FP16/int8+ calibration). All will get wrong results.  

And when using Polygraphy to analyze the layers of the onnx model and the engine model, the analysis results all passed the [instruction: polygraphy run UNet.onnx --load -- outputs UNet_onnx_out.json UNet_engine_out.json --onnxrt --trt --fail-fast】.

However, I saved the output of onnx inference and tensorrt inference as.npy files and compared the two files. Indeed, there are significant differences between them. The errors at many positions exceed 0.9.  

I sincerely hope to receive everyone's help！

**Comment by wangbiao0** [comment]
The following is the code of UNet. I compared the onnx operator support form of the tensorrt version I used and confirmed that both can be supported:


import torch.utils.data
import torch
import torch.nn as nn
from thop import profile

class DoubleConv(nn.Module):
"" "
Double Convolution Block
"" "

def __init__(self, in_ch, out_ch):
super(DoubleConv, self).__init__()

self.conv = nn.Sequential(
nn.Conv2d(in_ch, out_ch, kernel_size=3, stride=1, padding=1, bias=True),
nn.BatchNorm2d(out_ch),
# nn.ReLU(inplace=True),
nn.LeakyReLU(inplace=False),
nn.Conv2d(out_ch, out_ch, kernel_size=3, stride=1, padding=1, bias=True),
nn.BatchNorm2d(out_ch),
# nn.ReLU(inplace=True))
nn.LeakyReLU(inplace=False))

def forward(self, x):
x = self.conv(x)
return x


class up_conv(nn.Module):
"" "
Up Convolution Block
"" "

def __init__(self, in_ch, out_ch):
super(up_conv, self).__init__()
self.up = nn.Sequential(
nn.ConvTranspose2d(in_ch, out_ch, kernel_size=2, stride=2),
# nn.Upsample(scale_factor=2),
# nn.Upsample(scale_factor=2, mode='bilinear', align_corners=True),
# nn.Conv2d(in_ch, out_ch, kernel_size=3, stride=1, padding=1, bias=True),
# nn.BatchNorm2d(out_ch),
# nn.ReLU(inplace=True)
)

def forward(self, x):
x = self.up(x)
return x


class UNet(nn.Module):
"" "
UNet - Basic Implementation
"" "

def __init__(self, in_channels=3, num_classes=1):
super(UNet, self).__init__()

n1 = 32  #
filters = [n1, n1 * 2, n1 * 4, n1 * 8, n1 * 16]

self.Maxpool1 = nn.MaxPool2d(kernel_size=2, stride=2)
self.Maxpool2 = nn.MaxPool2d(kernel_size=2, stride=2)
self.Maxpool3 = nn.MaxPool2d(kernel_size=2, stride=2)
self.Maxpool4 = nn.MaxPool2d(kernel_size=2, stride=2)

self.Conv1 = DoubleConv(in_channels, filters[0])  # 3-32
self.Conv2 = DoubleConv(filters[0], filters[1])  # 32-64
self.Conv3 = DoubleConv(filters[1], filters[2])  # 64-128
self.Conv4 = DoubleConv(filters[2], filters[3])  # 128-256
self.Conv5 = DoubleConv(filters[3], filters[4])  # 256-512

self.Up5 = up_conv(filters[4], filters[3])  # 512-256
self.Up_conv5 = DoubleConv(filters[4], filters[3])  # 512-256

self.Up4 = up_conv(filters[3], filters[2])  # 256-128
self.Up_conv4 = DoubleConv(filters[3], filters[2])  # 256-128

self.Up3 = up_conv(filters[2], filters[1])  # 128-64
self.Up_conv3 = DoubleConv(filters[2], filters[1])  # 128-64

self.Up2 = up_conv(filters[1], filters[0])  # 64-32
self.Up_conv2 = DoubleConv(filters[1], filters[0])  # 64-32

self.Conv = nn.Conv2d(filters[0], num_classes, kernel_size=1, stride=1, padding=0)  # 32-1

# self.active = torch.nn.Sigmoid()

def forward(self, x):  # 3

e1 = self.Conv1(x)  # 32

e2 = self.Maxpool1(e1)
e2 = self.Conv2(e2)  # 64

e3 = self.Maxpool2(e2)
e3 = self.Conv3(e3)  # 128

e4 = self.Maxpool3(e3)
e4 = self.Conv4(e4)  # 256

e5 = self.Maxpool4(e4)
e5 = self.Conv5(e5)  # 512

d5 = self.Up5(e5)  # 256
d5 = torch.cat((e4, d5), dim=1)  # 256+256=512
d5 = self.Up_conv5(d5)  # 256

d4 = self.Up4(d5)  # 128
d4 = torch.cat((e3, d4), dim=1)  # 128+128=256
d4 = self.Up_conv4(d4)  # 128

d3 = self.Up3(d4)  # 64
d3 = torch.cat((e2, d3), dim=1)  # 64+64=128
d3 = self.Up_conv3(d3)  # 64

d2 = self.Up2(d3)  # 32
d2 = torch.cat((e1, d2), dim=1)  # 32+32=64
d2 = self.Up_conv2(d2)  # 32

out = self.Conv(d2)  # 1
#out = self.active(out)
return out


if __name__ == '__main__':
model = UNet(3, 3)
input_tensor = torch.randn(1, 3, 256, 256)
flops, params = profile(model, inputs=(input_tensor,))
print(f"FLOPs: {flops / 1e9:.2f} GFLOPs")
print(f"Parameters: {params / 1e6:.2f} M")

output = model(input_tensor)
print(f"Output shape: {output.shape}")
